lightwood/encoders/time_series/rnn.py

```diff
@@ -5,122 +5,171 @@
 from lightwood.helpers.device import get_devices
 from lightwood.encoders.encoder_base import EncoderBase
 
+import numpy as np
 import torch
 import torch.nn as nn
 from torch import optim
-import numpy as np
 
 
 class RnnEncoder(EncoderBase):
+    def __init__(self, encoded_vector_size=4, train_iters=75000, stop_on_error=0.8, learning_rate=0.01,
+                 is_target=False, ts_n_dims=1, dropout=0.2, weight_decay=1e-4, max_timesteps=64):
+        self.device, _ = get_devices()
 
-    def __init__(self, encoded_vector_size=4, train_iters=75000, stop_on_error=0.8, learning_rate=0.01, is_target=False):
         self._stop_on_error = stop_on_error
         self._learning_rate = learning_rate
         self._encoded_vector_size = encoded_vector_size
         self._train_iters = train_iters
         self._pytorch_wrapper = torch.FloatTensor
-        self._encoder = None
+        self._encoder = EncoderRNNNumerical(input_size=ts_n_dims, hidden_size=self._encoded_vector_size,
+                                            dropout=dropout).to(self.device)
+        self._decoder = DecoderRNNNumerical(output_size=ts_n_dims, hidden_size=self._encoded_vector_size,
+                                            dropout=dropout).to(self.device)
+        self.parameters = list(self._encoder.parameters()) + list(self._decoder.parameters())
+        self.optimizer = optim.AdamW(self.parameters, lr=self._learning_rate, weight_decay=weight_decay)
+        self.criterion = nn.MSELoss()
```

Comment:
Internal members, use `_` to declare unless I'm mistaken on them being internal only